torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -620,174 +750,128 @@ class IrParser {
             TensorView* running_var = nullptr;
             if (!node->input(4)->type()->isSubtypeOf(
                     static_cast<c10::TypePtr>(NoneType::get()))) {
-              running_var =
+              running_mean =
                   value_map[node->input(4)->unique()]->as<TensorView>();
             }
 
+            TensorView* save_mean = nullptr;
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            if (!node->input(5)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+              save_mean = value_map[node->input(5)->unique()]->as<TensorView>();
+            }
+
+            TensorView* save_invstd = nullptr;
             // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
-            // auto training = constant_as<bool>(node->input(5));
-            // TORCH_INTERNAL_ASSERT(
-            //     training.has_value(),
-            //     "The training (bool) parameter is required.");
-            // const bool kTraining = training.value();
+            if (!node->input(6)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              save_invstd =
+                  // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+                  value_map[node->input(6)->unique()]->as<TensorView>();
+            }
 
             // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
-            // auto momentum = constant_as<float>(node->input(6));
-            // TORCH_INTERNAL_ASSERT(
-            //     momentum.has_value(),
-            //     "The momentum (float) parameter is required.");
-            // const float kMomentum = momentum.value();
+            auto training = constant_as<bool>(node->input(7));
```

Comment:
If you expect this as constant, you should add a profiling rule for this. hint: `profileBool`

Reply:
How do I use `profileBool`?

Reply:
You just need to tell PE to profile certain arguments to the node as an typed data:
https://github.com/csarofeen/pytorch/blob/b08064db1e8aa1a0b431433c559449553a55992a/torch/csrc/jit/codegen/cuda/parser.cpp#L1692-L1710

graph_fuser.cpp would hook up profiling guard and convert the argument to constant for you automatically.